internal/services/compute/virtual_machine_run_command_resource.go

```diff
@@ -0,0 +1,702 @@
+	ScriptUriManagedIdentity []VirtualMachineRunCommandManagedIdentitySchema `tfschema:"script_uri_managed_identity"`
+}
+
+func (r VirtualMachineRunCommandResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return virtualmachineruncommands.ValidateVirtualMachineRunCommandID
+}
+
+func (r VirtualMachineRunCommandResource) ResourceType() string {
+	return "azurerm_virtual_machine_run_command"
+}
+
+func (r VirtualMachineRunCommandResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.VirtualMachineRunCommandName,
+		},
+
+		"virtual_machine_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateVirtualMachineID,
+		},
+
+		"source": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"command_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+						ExactlyOneOf: []string{
+							"source.0.command_id",
+							"source.0.script",
+							"source.0.script_uri",
+						},
+					},
+					"script": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+						ExactlyOneOf: []string{
+							"source.0.command_id",
+							"source.0.script",
+							"source.0.script_uri",
+						},
+					},
+					"script_uri": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.IsURLWithHTTPS,
+						ExactlyOneOf: []string{
+							"source.0.command_id",
+							"source.0.script",
+							"source.0.script_uri",
+						},
+					},
+					"script_uri_managed_identity": {
+						Type:      pluginsdk.TypeList,
+						Optional:  true,
+						Sensitive: true,
+						MaxItems:  1,
+						RequiredWith: []string{
+							"source.0.script_uri",
+						},
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"client_id": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									Sensitive:    true,
+									ValidateFunc: validation.StringIsNotEmpty,
+									ConflictsWith: []string{
+										"source.0.script_uri_managed_identity.0.object_id",
+									},
+								},
+								"object_id": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									Sensitive:    true,
+									ValidateFunc: validation.StringIsNotEmpty,
+									ConflictsWith: []string{
+										"source.0.script_uri_managed_identity.0.client_id",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"async_execution_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
```

Comment:
This would violate a key principal of Terraform - that we don't mark a resource as `Succeeded` until it's finished provisioning - so we'd need to remove this.

Reply:
Make sense, now the property is hide in TF schema. And we set the flag to `false` in request.

Reply:
@tombuildsstuff async_execution is pretty important for this resource because if you're setting up a sql vm, it could take hours to restore backups. That's why the AzureRM API supports async_execution. As currently implemented, if the script takes hours to run then terraform has to run for hours, eating up minutes on our CI/CD system and tying up agents. To me, it seems perfectly valid to consider the resource provisioned if the run command is created, even if it is still running.